backup-stores/gcs/src/main/java/io/camunda/zeebe/backup/gcs/ManifestManager.java

```diff
@@ -137,5 +154,41 @@ public void markAsFailed(final BackupIdentifier id, final String failureReason)
     }
   }
 
+  public Collection<Manifest> listManifests(final BackupIdentifierWildcard wildcard) {
+    final var spliterator =
+        Spliterators.spliteratorUnknownSize(
+            client
+                .list(bucketInfo.getName(), BlobListOption.prefix(prefix))
+                .iterateAll()
+                .iterator(),
+            Spliterator.IMMUTABLE);
+    return StreamSupport.stream(spliterator, false)
+        .filter(filterBlobsByWildcard(wildcard))
+        .parallel()
+        .map(Blob::getContent)
+        .map(
+            content -> {
+              try {
+                return MAPPER.readValue(content, Manifest.class);
+              } catch (final IOException e) {
+                throw new RuntimeException(e);
+              }
+            })
+        .toList();
+  }
+
+  private Predicate<Blob> filterBlobsByWildcard(final BackupIdentifierWildcard wildcard) {
```

Comment:
:wrench: Suggestion to improve performance and reduce a bit the complexity, instead of using a pattern to match the values, extracting them and then matching the right values proposal would be to compile an pattern based on the given wildcard: e.g. `(?\d+)/(?\d+)/1` if node id 1 is given. This allows to directly filter and match names.